osu.Game/Beatmaps/BeatmapInfo.cs

```diff
@@ -125,9 +125,10 @@ public BeatmapOnlineStatus Status
         /// <summary>
         /// Reset any fetched online linking information (and history).
         /// </summary>
-        public void ResetOnlineInfo()
+        public void ResetOnlineInfo(bool resetOnlineId = true)
```

Comment:
I have a 99.999999% confidence this will never again be called with `resetOnlineId: false` outside of the migration this PR is adding, and as such I would have just copied the logic across because I utterly detest boolean parameters like this that basically turn off some of the method, but I don't feel strongly enough about it to demand it changed